optimum/habana/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -378,3 +385,10 @@ def prepare_inputs_for_generation(
             "past_key_values": past_key_values,
             "token_idx": token_idx,
         }
+
+
+def apply_customized_rope(q, k, cos, sin, position_ids):
+    if q.device.type == "hpu" and FusedRoPE:
```

Comment:
```suggestion
    if FusedRoPE:
```
q.device should always be 'hpu', means not support CPU or CUDA

Reply:
this optimum-habana script should be able to run on CPU, right?

Reply:
Discussed with Feiyue offline, we decided to keep device type check in the source code.